### PR TITLE
Ship and autoupdate to osquery app bundle

### PR DESCRIPTION
**Pull request by RebeccaMahany**
Requires https://github.com/kolide/endpoint-releaser/pull/115

* Updates package-builder to include the osquery app bundle in the .pkg
* Updates new autoupdater to find executable within osquery.app directory when downloaded (the legacy autoupdate code in `findnew.go` already correctly handles finding the path to osquery.app's executable)
* Adds tests to `Test install macOS` job

<details>
<summary>Testing notes</summary>

* Tested building a pkg with package-builder -- test steps largely mirror the ones documented in https://github.com/kolide/launcher/pull/934
* Moved an osquery app bundle into the legacy updates directory to ensure that launcher could find it there
* Moved an osquery app bundle into the new updates directory to ensure that launcher could find it there

</details>

### CODE REVIEW

**Review by James-Pickett — APPROVED** [review]
Nice

File: pkg/autoupdate/tuf/ci/tuf_server.go [hunk 1]
```diff
@@ -163,11 +163,15 @@ func compress(t *testing.T, outFileName string, outFileDir string, targetDir str
...
 
 	srcFilePath := binary
-	if binary == "launcher" && runtime.GOOS == "darwin" {
-		srcFilePath = filepath.Join("Kolide.app", "Contents", "MacOS", binary)
+	if runtime.GOOS == "darwin" {
+		appBundleName := "Kolide.app"
+		if binary == "osqueryd" {
+			appBundleName = "osquery.app"
+		}
+		srcFilePath = filepath.Join(appBundleName, "Contents", "MacOS", binary)
```
Comment (directionless) [inline pkg/autoupdate/tuf/ci/tuf_server.go:171]:
```suggestion
		srcFilePath = filepath.Join(appBundleName, "Contents", "MacOS", binary)
		
```

**Review by directionless — APPROVED** [review]
Looks like a small addition to the launcher app bundle work. Nice

File: pkg/autoupdate/tuf/ci/tuf_server.go (hunk 1 above)
Comment (RebeccaMahany) [reply pkg/autoupdate/tuf/ci/tuf_server.go:171]: Sorry, I applied this and then realized it resulted in two empty newlines, so I've undone the change. Was there something else you wanted to update here?

Comment (RebeccaMahany) [reply pkg/autoupdate/tuf/ci/tuf_server.go:171]: There already was one, that's why lint failed

Comment (RebeccaMahany) [reply pkg/autoupdate/tuf/ci/tuf_server.go:171]: Do you want a new line before 171 and not after it?

Comment (directionless) [reply pkg/autoupdate/tuf/ci/tuf_server.go:172]: Before 171, but it doesn't matter much. My thinking is that you have:
```
		appBundleName := "Kolide.app"

		if binary == "osqueryd" {
			appBundleName = "osquery.app"
		}
		srcFilePath = filepath.Join(appBundleName, "Contents", "MacOS", binary)
```

which grouped things into:
1. set appBundleName
2. change appBundleName, and set srcFilePath

and that just seems weird to me. So I'd shuffle the newlines to feel more grouped. But 🤷

**Review by directionless — APPROVED** [review]

**Pull request merged**